c2corg_api/scripts/migration/documents/articles.py

```diff
@@ -0,0 +1,118 @@
+from c2corg_api.models.article import Article, ArchiveArticle, ARTICLE_TYPE
+from c2corg_api.models.document import DocumentLocale, ArchiveDocumentLocale, \
+    DOCUMENT_TYPE
+from c2corg_api.scripts.migration.documents.document import MigrateDocuments
+from c2corg_api.scripts.migration.documents.routes import MigrateRoutes
+
+
+class MigrateArticles(MigrateDocuments):
+
+    def get_name(self):
+        return 'articles'
+
+    def get_model_document(self, locales):
+        return DocumentLocale if locales else Article
+
+    def get_model_archive_document(self, locales):
+        return ArchiveDocumentLocale if locales else ArchiveArticle
+
+    def get_document_geometry(self, document_in, version):
+        return dict(
+            document_id=document_in.id,
+            id=document_in.id,
+            version=version,
+            geom=document_in.geom
+        )
+
+    def get_count_query(self):
+        return (
+            ' select count(*) '
+            ' from app_articles_archives aa join articles t on aa.id = t.id '
+            ' where t.redirects_to is null;'
+        )
+
+    def get_query(self):
+        return (
+            ' select '
+            '   aa.id, aa.document_archive_id, aa.is_latest_version, '
+            '   aa.is_protected, aa.redirects_to, '
+            '   ST_Force2D(ST_SetSRID(aa.geom, 3857)) geom, aa.elevation, '
+            '   aa.categories, aa.activities, aa.article_type '
+            ' from app_articles_archives aa join articles t on aa.id = t.id '
+            ' where t.redirects_to is null '
+            ' order by aa.id, aa.document_archive_id;'
+        )
+
+    def get_count_query_locales(self):
+        return (
+            ' select count(*) '
+            ' from app_articles_i18n_archives aa '
+            '   join articles t on aa.id = t.id '
+            ' where t.redirects_to is null;'
+        )
+
+    def get_query_locales(self):
+        return (
+            ' select '
+            '    aa.id, aa.document_i18n_archive_id, aa.is_latest_version, '
+            '    aa.culture, aa.name, aa.description '
+            ' from app_articles_i18n_archives aa '
+            '   join articles t on aa.id = t.id '
+            ' where t.redirects_to is null '
+            ' order by aa.id, aa.culture, aa.document_i18n_archive_id;'
+        )
+
+    def get_document(self, document_in, version):
+        categories = self.convert_types(
+            document_in.categories, MigrateArticles.article_categories)
+        print(categories)
+        if 'draft' in categories:
+            default_quality = 'draft'
+            categories.remove('draft')
+        else:
+            default_quality = 'medium'
```

Comment:
You could use the constant `DEFAULT_QUALITY` in `c2corg_api.scripts.migration.documents.document`. If we decide to change the default quality, we will not forget to change it here.